### PR TITLE
Update NNPI Backend to v0.5.0.8

### PR DESCRIPTION
**Pull request by omromano**
Summary: Update of NNPI Backend to v0.5.0.8 branched out of commit 24df161c4d0eb716473eac43673b423777a9ab48.

### CODE REVIEW

**Comment by jsubag** [comment]
@arunm-git @jfix71 Please take a look

File: lib/Backends/NNPI/InferenceContext.cpp [hunk 1]
```diff
@@ -0,0 +1,433 @@
...
+  }
+  TRACE_EVENT_BEGIN(ctx->getTraceContext(), TraceLevel::COPY,
+                    TRACING_PRE_PROCESS);
+
+  // Pre-inference
+  auto &phTensorMap = bindings.pairs();
+  std::vector<void *> rawInputs, rawOutputs;
+  for (auto &in : inputResources_) {
+    if (in->GetUsage() != NNPIResource::ResourceUsage::StaticInputResource) {
+      Placeholder *ph = bindings.getPlaceholderByName(in->GetName());
```
Comment (yinghai) [inline lib/Backends/NNPI/InferenceContext.cpp]: This seems to be a regression from recent positional bindings as here we are doing string lookup per inference per input again.

Comment (jsubag) [reply lib/Backends/NNPI/InferenceContext.cpp]: Yes, we can apply a similar change as in #4147 here.

**Comment by jsubag** [comment]
The refactor adds no new features.
It's a cleaner design and a step towards the context binding mechanism. In also gives some perf benefit in the pre/post processing before/after inferences.

File: lib/Backends/NNPI/InferenceContext.cpp (hunk 1 above)
Comment (yinghai) [reply lib/Backends/NNPI/InferenceContext.cpp]: Should we do it in this PR or a follow-up?

Comment (jsubag) [reply lib/Backends/NNPI/InferenceContext.cpp]: Will try to do that on Sunday - otherwise can do it later in the week.

**Comment by yinghai** [comment]
@jsubag I'm sendng https://github.com/pytorch/glow/pull/4194 for the follow-up.

**Pull request closed without merging**